### PR TITLE
update envoy to 1.14.1

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

- update Wasm to use the new declare/get property
- fix tests
- add a test using peer info

### CODE REVIEW

File: extensions/stackdriver/stackdriver.cc [hunk 1]
```diff
@@ -346,8 +344,7 @@ inline bool StackdriverRootContext::enableEdgeReporting() {
 
 bool StackdriverRootContext::shouldLogThisRequest() {
   std::string shouldLog = "";
-  if (!getValue({"filter_state", ::Wasm::Common::kAccessLogPolicyKey},
-                &shouldLog)) {
+  if (!getValue({::Wasm::Common::kAccessLogPolicyKey}, &shouldLog)) {
```
Comment (bianpengyuan) [inline extensions/stackdriver/stackdriver.cc:347]: I don't see this was declared as property anywhere. We should have test for this, wondering why test does not fail.

File: src/envoy/tcp/metadata_exchange/metadata_exchange.cc [hunk 2]
```diff
@@ -269,35 +270,55 @@ void MetadataExchangeFilter::tryReadProxyData(Buffer::Instance& data) {
...
+      reinterpret_cast<const char*>(fbb.GetBufferPointer()), fbb.GetSize()));
+
+  auto key = config_->filter_direction_ == FilterDirection::Downstream
+                 ? ::Wasm::Common::kDownstreamMetadataKey
+                 : ::Wasm::Common::kUpstreamMetadataKey;
   read_callbacks_->connection().streamInfo().filterState()->setData(
-      key,
-      std::make_unique<::Envoy::Extensions::Common::Wasm::WasmState>(value),
-      StreamInfo::FilterState::StateType::Mutable,
+      absl::StrCat("wasm.", key), std::move(state),
```
Comment (bianpengyuan) [inline src/envoy/tcp/metadata_exchange/metadata_exchange.cc:306]: Why attach the prefix here? Is it that Wasm runtime will attach the `wasm.` prefix by default so here it is to make the behavior consistent?

File: extensions/common/context.h [hunk 3]
```diff
@@ -30,20 +30,16 @@ using StringView = absl::string_view;
 // Node metadata
 constexpr StringView WholeNodeKey = ".";
 
-constexpr StringView kUpstreamMetadataIdKey =
-    "envoy.wasm.metadata_exchange.upstream_id";
-constexpr StringView kUpstreamMetadataKey =
-    "envoy.wasm.metadata_exchange.upstream";
+constexpr StringView kUpstreamMetadataIdKey = "upstream_peer_id";
```
[the remark this answers is not in the document]
Comment (kyessenov) [reply extensions/common/context.h:33]: We want to hide the implementation detail of filter_state from Wasm ABI. If we had dots there, then it's difficult to guess whether the dot belongs to the name of the filter state key, or a field traversal. Using a flat namespace `wasm.variable_name` fits most cases for now. The prefix is to make sure we don't collide with existing `envoy.*` variables.

File: WORKSPACE [hunk 4]
```diff
@@ -37,18 +37,18 @@ bind(
 # 1. Determine SHA256 `wget https://github.com/envoyproxy/envoy-wasm/archive/$COMMIT.tar.gz && sha256sum $COMMIT.tar.gz`
 # 2. Update .bazelversion, envoy.bazelrc and .bazelrc if needed.
 #
-# envoy-wasm commit time: Wed Mar 20 20:46:00 2020 -0700
-ENVOY_SHA = "d29f7a659ba736aab97697a7bcfc69a71bc66b66"
+# envoy-wasm commit time: 4/20/20
```
[the remark this answers is not in the document]
Comment (bianpengyuan) [reply WORKSPACE:40]: actually nvm. I think we might be going to switch back to envoy-wasm some time later, so let's keep it that way.

File: extensions/stackdriver/stackdriver.cc (hunk 1 above)
Comment (kyessenov) [reply extensions/stackdriver/stackdriver.cc:347]: It's set by access log policy extension. Declaration is not necessary, it'll default to a default declaration https://github.com/envoyproxy/envoy-wasm/commit/80e1a645c25f6243bfacf0eb5b9e96eb2e767aff#diff-965399965695ac238aae83e5d9bd5666R42. Get/set/declare now all use the same identifier for consistency.

File: src/envoy/tcp/metadata_exchange/metadata_exchange.cc (hunk 2 above)
Comment (kyessenov) [reply src/envoy/tcp/metadata_exchange/metadata_exchange.cc:306]: Yes. We should really be using Wasm ABI for this, so there's a lot of boilerplate.

File: extensions/stackdriver/stackdriver.cc (hunk 1 above)
Comment (bianpengyuan) [reply extensions/stackdriver/stackdriver.cc:347]: I see. `property` declaration is only needed for CEL expression. Find a typo during code reading: https://github.com/envoyproxy/envoy-wasm/blob/f80bba648062e95b98497b84ba20a8165828b082/source/extensions/common/wasm/context.cc#L428

Comment (kyessenov) [reply extensions/stackdriver/stackdriver.cc:347]: @jplevyak might want to fix the macro pollution ^^^

Comment (kyessenov) [reply extensions/stackdriver/stackdriver.cc:347]: getProperty should also be able to read inside buffers without using CEL compiler.

File: extensions/stackdriver/edges/mesh_edges_service_client.cc [hunk 5]
```diff
@@ -79,9 +79,11 @@ void MeshEdgesServiceClientImpl::reportTrafficAssertions(
   LOG_TRACE("mesh edge services client: sending request '" +
             request.DebugString() + "'");
 
-  context_->grpcSimpleCall(
-      grpc_service_, kMeshEdgesService, kReportTrafficAssertions, request,
-      kDefaultTimeoutMillisecond, success_callback_, failure_callback_);
+  HeaderStringPairs initial_metadata;
+  context_->grpcSimpleCall(grpc_service_, kMeshEdgesService,
```
Comment (mandarjog) [inline extensions/stackdriver/edges/mesh_edges_service_client.cc:83]: Is this because of an underlying API change?

Comment (kyessenov) [reply extensions/stackdriver/edges/mesh_edges_service_client.cc:83]: Yeah, the initial metadata callback was not usable until this.

**Pull request merged**